File: mshift-api/src/main/java/com/moneyshift/api/service/TransactionService.java
package com.moneyshift.api.service;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.moneyshift.api.model.TransactionData;
import com.moneyshift.api.model.TransactionData.AccountInfo;
import lombok.extern.slf4j.Slf4j;
import org.springframework.core.io.ClassPathResource;
import org.springframework.stereotype.Service;

import java.io.IOException;

@Service
@Slf4j
public class TransactionService {
    
    private final ObjectMapper objectMapper = new ObjectMapper();
    
    public TransactionData getBankATransactions() {
        try {
            // 개발용: JSON 파일에서 데이터 로드
            ClassPathResource resource = new ClassPathResource("data/bank-a-transaction.json");
            TransactionData data = objectMapper.readValue(resource.getInputStream(), TransactionData.class);
            
            log.info("Successfully loaded {} transactions from JSON file", 
                    data.getTransactions() != null ? data.getTransactions().size() : 0);
            
            return data;
        } catch (IOException e) {
            log.error("Failed to load transaction data from JSON file", e);
            
            // 실패 시 기본 데이터 반환
            return createDefaultTransactionData();
        }
    }
    
    private TransactionData createDefaultTransactionData() {
        TransactionData defaultData = new TransactionData();
        
        // 기본 계좌 정보 설정
        AccountInfo accountInfo = new AccountInfo();
        accountInfo.setBankName("기업은행");
        accountInfo.setAccountNumber("****-****-****-1234");
        accountInfo.setBalance("0");
        accountInfo.setCurrency("원");
        
        defaultData.setAccountInfo(accountInfo);
        defaultData.setTransactions(java.util.Collections.emptyList());
        
        return defaultData;
    }
}

File: mshift-trojan-app-fixed/src/services/TransactionService.ts
import { TransactionData } from '../types/Transaction';
import { apiCall, API_CONFIG } from '../config/api';

class TransactionService {
  private static instance: TransactionService;
  private transactionData: TransactionData | null = null;

  private constructor() {}

  public static getInstance(): TransactionService {
    if (!TransactionService.instance) {
      TransactionService.instance = new TransactionService();
    }
    return TransactionService.instance;
  }

  public async loadTransactionData(): Promise<TransactionData> {
    try {
      console.log('Loading transaction data from API...');
      
      // API에서 거래 데이터 로드
      const response = await apiCall(API_CONFIG.ENDPOINTS.TRANSACTIONS_BANK_A);
      
      if (response.ok) {
        const data = await response.json();
        this.transactionData = data;
        console.log('Transaction data loaded successfully from API.');
        console.log('Account info:', data.accountInfo);
        console.log('Transactions count:', data.transactions?.length || 0);
        return data;
      } else {
        console.error('Failed to load transaction data from API:', response.status, response.statusText);
        throw new Error(`API call failed with status: ${response.status}`);
      }
    } catch (error) {
      console.error('Failed to load transaction data from API:', error);
      
      // API 실패 시에도 에러를 던져서 UI에서 에러 처리하도록 함
      throw error;
    }
  }

  public async refreshData(): Promise<TransactionData> {
    this.transactionData = null;
    return await this.loadTransactionData();
  }

  public getTransactionData(): TransactionData | null {
    return this.transactionData;
  }
}

export default TransactionService;